Sources/Snapshot.swift

```diff
@@ -0,0 +1,38 @@
+/// Represents a snapshot of a collection.
+///
+/// The changeset associated with the first snapshot received by any given observation
+/// is generally irrelevant. Observations should special case the first snapshot as a
+/// complete replacement, or whatever semantic that fits their purpose.
+///
+/// The previous version of the collection is unavailable on the initial snapshot,
+/// since there is no history to refer to.
+public struct Snapshot<Collection: Swift.Collection>: SnapshotProtocol {
+	/// The previous version of the collection, or `nil` if `self` is an initial snapshot.
+	public let previous: Collection?
+
+	/// The current version of the collection.
+	public let current: Collection
+
+	/// The changeset which, when applied on `previous`, reproduces `current`.
+	public let changeset: Changeset
```

Comment:
Hmm, it would also be an issue at property level, if `PropertyProtocol.diff` is going to return a conventional `Property<Snapshot>`.

TBH packing as an enum makes sense only if we limit the diffing to _properties_, and guarantees that the producer always sends out an `initial` snapshot, akin to the "guaranteed current value".